src/test/components/ErrorBoundary.test.js

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import * as React from 'react';
+import { render } from 'react-testing-library';
+
+import { ErrorBoundary } from '../../components/app/ErrorBoundary';
+import { withAnalyticsMock } from '../fixtures/mocks/analytics';
+
+describe('app/ErrorBoundary', function() {
+  const childComponentText = 'This is a child component';
+  const friendlyErrorMessage = 'Oops, there was an error';
+  const technicalErrorMessage = 'This is an error.';
+  const ThrowingComponent = () => {
+    throw new Error(technicalErrorMessage);
+  };
+
+  function setupComponent(childComponent) {
+    const spy = jest.spyOn(console, 'error').mockImplementation(() => {});
+    const results = render(
+      <ErrorBoundary message={friendlyErrorMessage}>
+        {childComponent}
+      </ErrorBoundary>
+    );
+    return { spy, ...results };
+  }
+
+  it('shows the component children when there is no error', () => {
+    const { getByText } = setupComponent(childComponentText);
+    expect(getByText(childComponentText)).toBeTruthy();
+  });
+
+  it('shows the error message children when the component throws error', () => {
+    jest.spyOn(console, 'error').mockImplementation(() => {});
+    const { getByText } = setupComponent(<ThrowingComponent />);
+    expect(getByText(friendlyErrorMessage)).toBeTruthy();
+  });
+
+  it('surfaces the error via console.error', () => {
+    const { spy } = setupComponent(<ThrowingComponent />);
+    expect(spy).toHaveBeenCalled();
```

Comment:
optional: you can do `expect(console.error).toHaveBeenCalled()` and avoid storing the spy. I believe this is more understandable, because the variable name `spy` isn't very descriptive.